### PR TITLE
executor, tests: move some test cases from `executor_test.go` to `integrationtest`

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: ref #47076

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: tests/integrationtest/t/executor/executor.test [hunk 1]
```diff
@@ -0,0 +1,676 @@
...
+insert t values (3.3, 2.7);
+select * from t where a > 1.5;
+select * from t where b > 1.5;
+drop table if exists t;
+create table t (a time(3), b time, index idx_a (a));
+insert t values ('11:11:11', '11:11:11');
+insert t values ('11:11:12', '11:11:12');
+insert t values ('11:11:13', '11:11:13');
+select * from t where a > '11:11:11.5';
+select * from t where b > '11:11:11.5';
```
Comment (Defined2014) [inline tests/integrationtest/t/executor/executor.test:509]: need restore `sql_mode = default`

File: tests/integrationtest/t/executor/executor.test [hunk 2]
```diff
@@ -0,0 +1,676 @@
...
+insert t values ('ab123', 2);
+select * from t where a like 'ab%';
+select * from t where a like 'ab_12';
+drop table if exists t;
+create table t (a int primary key);
+insert t values (1);
+insert t values (2);
+select * from t where not (a = 1);
+select * from t where not(not (a = 1));
+select * from t where not(a != 1 and a != 2);
```
Comment (Defined2014) [inline tests/integrationtest/t/executor/executor.test:492]: need restore sql_mode = default

File: tests/integrationtest/t/executor/executor.test [hunk 3]
```diff
@@ -0,0 +1,676 @@
...
+select b from t where a > 0 limit 1, 4;
+set @@tidb_init_chunk_size=2;
+select * from t where a > 0 limit 2, 1;
+select * from t where a > 0 limit 2, 2;
+select * from t where a > 0 limit 2, 3;
+select * from t where a > 0 limit 2, 4;
+select a from t order by a limit 2, 1;
+select b from t order by a limit 2, 2;
+select a from t order by a limit 2, 3;
+select b from t order by a limit 2, 4;
```
Comment (Defined2014) [inline tests/integrationtest/t/executor/executor.test:229]: need restore tidb_init_check_size=default;

File: tests/integrationtest/t/executor/executor.test [hunk 4]
```diff
@@ -0,0 +1,676 @@
...
+
+# TestUnsignedPKColumn
+drop table if exists t;
+create table t (a int unsigned primary key, b int, c int, key idx_ba (b, c, a));
+insert t values (1, 1, 1);
+select * from t;
+update t set c=2 where a=1;
+select * from t where b=1;
+
+# TestMultiUpdate
```
Comment (Defined2014) [inline tests/integrationtest/t/executor/executor.test:62]: Miss delete function in `_test.go`

File: tests/integrationtest/t/executor/executor.test [hunk 5]
```diff
@@ -0,0 +1,676 @@
...
+INSERT INTO test_mu VALUES (1, 2, 3) ON DUPLICATE KEY UPDATE c = 2, b = c+5;
+SELECT * FROM test_mu ORDER BY a;
+UPDATE test_mu SET b = 0, c = b WHERE a = 4;
+SELECT * FROM test_mu ORDER BY a;
+UPDATE test_mu SET c = 8, b = c WHERE a = 4;
+SELECT * FROM test_mu ORDER BY a;
+UPDATE test_mu SET c = b, b = c WHERE a = 7;
+SELECT * FROM test_mu ORDER BY a;
+
+# TestGeneratedColumnPointGet
```
Comment (Defined2014) [inline tests/integrationtest/t/executor/executor.test:76]: ditto

File: tests/integrationtest/t/executor/executor.test [hunk 6]
```diff
@@ -0,0 +1,676 @@
...
+select * from tu where e = 4;
+update tu set a = a + 1, b = b + 1 where c = 11;
+select * from tu for update;
+select * from tu where a = 6;
+select * from tu where c in (5, 6, 13);
+select b, c, e, d from tu where c = 13;
+select a, e, d from tu where c in (5, 6, 13);
+drop table if exists tu;
+
+# TestUnionAutoSignedCast
```
Comment (Defined2014) [inline tests/integrationtest/t/executor/executor.test:100]: ditto

File: tests/integrationtest/t/executor/executor.test [hunk 7]
```diff
@@ -0,0 +1,676 @@
...
+create table t5 (id int, v bigint unsigned);
+create table t6 (id int, v decimal);
+create table t7 (id int, v bigint);
+insert into t5 values (1, 1);
+insert into t6 values (2, -1);
+insert into t7 values (3, -1);
+select id, v from t5 union select id, v from t6 order by id;
+select id, v from t5 union select id, v from t7 union select id, v from t6 order by id;
+
+# TestDeletePartition
```
Comment (Defined2014) [inline tests/integrationtest/t/executor/executor.test:127]: ditto

File: tests/integrationtest/t/executor/executor.test [hunk 8]
```diff
@@ -0,0 +1,676 @@
...
+ );
+insert into t1 values (1),(11),(21),(31);
+delete from t1 partition (p4);
+select * from t1 order by a;
+delete from t1 partition (p0) where a > 10;
+select * from t1 order by a;
+delete from t1 partition (p0,p1,p2);
+select * from t1;
+
+# TestAlterTableComment
```
Comment (Defined2014) [inline tests/integrationtest/t/executor/executor.test:144]: ditto

File: tests/integrationtest/t/executor/executor.test [hunk 9]
```diff
@@ -0,0 +1,676 @@
...
+drop table if exists t_1;
+create table t_1 (c1 int, c2 int, c3 int default 1, index (c1)) comment = 'test table';
+alter table `t_1` comment 'this is table comment';
+select table_comment from information_schema.tables where table_name = 't_1';
+alter table `t_1` comment 'table t comment';
+select table_comment from information_schema.tables where table_name = 't_1';
+
+# TestExecutorEnum
```
Comment (Defined2014) [inline tests/integrationtest/t/executor/executor.test:152]: ditto

File: tests/integrationtest/t/executor/executor.test [hunk 10]
```diff
@@ -0,0 +1,676 @@
...
+select c + 1 from t where c = 2;
+delete from t;
+insert into t values ();
+insert into t values (null), ('1');
+select c + 1 from t where c = 1;
+delete from t;
+insert into t values(1), (2), (3);
+select * from t where c;
+
+# TestExecutorSet
```
Comment (Defined2014) [inline tests/integrationtest/t/executor/executor.test:166]: ditto

File: tests/integrationtest/t/executor/executor.test (hunk 1 above)
Comment (SeaRise) [reply tests/integrationtest/t/executor/executor.test:514]: ok, added sql like
```
set @tmp=(select @@time_zone);
set @@time_zone = '+10:00';
...
set @@time_zone = @tmp;
```

File: tests/integrationtest/t/executor/executor.test (hunk 2 above)
Comment (SeaRise) [reply tests/integrationtest/t/executor/executor.test:495]: ditto

File: tests/integrationtest/t/executor/executor.test (hunk 1 above)
Comment (Defined2014) [reply tests/integrationtest/t/executor/executor.test:514]: maybe `set @@time_zone = default` is enough. But this one is better.

File: tests/integrationtest/t/executor/executor.test (hunk 4 above)
Comment (SeaRise) [reply tests/integrationtest/t/executor/executor.test:62]: ok, deleted.

File: tests/integrationtest/t/executor/executor.test (hunk 5 above)
Comment (SeaRise) [reply tests/integrationtest/t/executor/executor.test:76]: ok, deleted.

File: tests/integrationtest/t/executor/executor.test (hunk 6 above)
Comment (SeaRise) [reply tests/integrationtest/t/executor/executor.test:100]: ok, deleted.

File: tests/integrationtest/t/executor/executor.test (hunk 7 above)
Comment (SeaRise) [reply tests/integrationtest/t/executor/executor.test:127]: ok, deleted.

File: tests/integrationtest/t/executor/executor.test (hunk 8 above)
Comment (SeaRise) [reply tests/integrationtest/t/executor/executor.test:144]: ok, deleted.

File: tests/integrationtest/t/executor/executor.test (hunk 9 above)
Comment (SeaRise) [reply tests/integrationtest/t/executor/executor.test:152]: ok, deleted.

File: tests/integrationtest/t/executor/executor.test (hunk 10 above)
Comment (SeaRise) [reply tests/integrationtest/t/executor/executor.test:166]: ok, deleted.

File: tests/integrationtest/t/executor/executor.test (hunk 1 above)
Comment (bb7133) [reply tests/integrationtest/t/executor/executor.test:514]: > maybe `set @@time_zone = default` is enough. But this one is better.

Why this one is better? IMO `set @@time_zone = default` seems easier and straightfoward.

**Comment by bb7133** [comment]
Some of test cases can be ported with some tiny updates, for example `TestSignedCommonHandle` we could convert:
```
tk.Session().GetSessionVars().EnableClusteredIndex = variable.ClusteredIndexDefModeOn
```
to:
```
set @@tidb_enable_clustered_index = on
```
and this is actually the better way to change clustered index mode in a test case IMO.

However, we don't necessarily include them in this PR.

**Review by bb7133 — APPROVED** [review]
LGTM

**Review by Defined2014 — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

File: tests/integrationtest/t/executor/executor.test (hunk 1 above)
Comment (SeaRise) [reply tests/integrationtest/t/executor/executor.test:514]: Okay, I'll change it to `set @@time_zone = default`

Comment (SeaRise) [reply tests/integrationtest/t/executor/executor.test:511]: done

**Pull request merged**